knowledge-base/grid-custom-filter-menu.md

```diff
@@ -0,0 +1,201 @@
+---
+title: How to Refresh Filter Menu Context After Programmatic Changes
+description: Learn how to refresh filter menu context after programmatic changes with a custom component
+type: how-to
+page_title: How to Refresh Filter Menu Context After Programmatic Changes
+slug: grid-kb-custom-filter-menu
+tags: grid, blazor, custom, filter, filtermenu
+res_type: kb
+ticketid: 1669381
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Grid for Blazor</td>
+	    </tr>
+    </tbody>
+</table>
+
+## Description
+
+When using the TelerikGrid in Blazor applications, it's common to implement custom `FilterMenu` for enhanced filtering capabilities. However, integrating an [AutoComplete]({%slug autocomplete-overview%}) component with `StringOperators` within a [`FilterMenuTemplate`]({%slug grid-templates-filter%}#filter-menu-template) for dynamic filtering based on user input does not work as expected. The AutoComplete component does not consider the selection from the dropdown list of `StringOperators` within the `FilterMenuTemplate`. 
+
+This knowledge base article also answers the following questions:
+- How to integrate AutoComplete with `StringOperators` in a TelerikGrid `FilterMenuTemplate`?
+- How to refresh a custom `FilterMenuTemplate` in TelerikGrid?
```

Comment:
The content of every template is "custom", so no need to explicitly mention this.

```suggestion
- How to refresh the contents of `FilterMenuTemplate` in a TelerikGrid for Blazor?
```